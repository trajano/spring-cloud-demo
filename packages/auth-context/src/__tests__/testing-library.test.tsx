import {
  act,
  cleanup,
  fireEvent,
  render,
  screen,
} from '@testing-library/react-native';
import React, { useCallback, useState } from 'react';
import { Pressable, Text, TextInput, View } from 'react-native';

beforeEach(() => {
  jest.useFakeTimers();
});
afterEach(() => {
  jest.useRealTimers();
});
afterEach(cleanup);

function Example() {
  const [name, setUser] = useState('');
  const [show, setShow] = useState(false);

  const onPress = useCallback(() => {
    // let's pretend this is making a server request, so it's async
    // (you'd want to mock this imaginary request in your unit tests)...
    setTimeout(() => {
      setShow(true);
    }, Math.floor(Math.random() * 200));
  }, [setShow]);
  return (
    <View>
      <TextInput value={name} onChangeText={setUser} testID="input" />
      <Pressable onPress={onPress}>
        <Text>Print Username</Text>
      </Pressable>
      {show && <Text testID="printed-username">{name}</Text>}
    </View>
  );
}

test('examples of some things', () => {
  const expectedUsername = 'Ada Lovelace';

  render(<Example />);

  fireEvent.changeText(screen.getByTestId('input'), expectedUsername);
  fireEvent.press(screen.getByText('Print Username'));
  act(() => jest.advanceTimersToNextTimer());
  const usernameOutput = screen.getByTestId('printed-username');

  // Using `toHaveTextContent` matcher from `@testing-library/jest-native` package.
  expect(usernameOutput).toHaveTextContent(expectedUsername);

  expect(screen.toJSON()).toMatchSnapshot();
});
