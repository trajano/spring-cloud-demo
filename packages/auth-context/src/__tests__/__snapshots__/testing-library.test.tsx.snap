// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples of some things 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    testID="input"
    value="Ada Lovelace"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      Print Username
    </Text>
  </View>
  <Text
    testID="printed-username"
  >
    Ada Lovelace
  </Text>
</View>
`;
